src/com/mercado/model/BaseEntity.java

```diff
@@ -0,0 +1,44 @@
+package com.mercado.model;
+
+import java.util.Objects;
+
+public abstract class BaseEntity {
+    private Integer id;
```

Comment:
What do you think about using 
`Long` for ids?

Reply:
I think 2,147,483,647 records is enough for mayority  cases, but only in the scenario where we use something like a chat or record for every fly born in the world would we need
 more space.